### DISCUSSION TITLE
Error creating datasource: Unable to connect: connect ECONNREFUSED 127.0.0.1:3306 this error pops up when i try to add.

### DISCUSSION THREAD

**Discussion by Townieeee** [Budibase/budibase#13565] [Help] [answered]
I saw someone had asked and resolved this, but I was not as lucky.
I have a working and running MYSQL setup with a TestDB and 1 table on 127.0.0.1 with user set as root and pw set as root. (port 3306).
From Budibase I created a test App and tried to connect to MYSQL with my above parameters (all on my Windows 10 laptop).
I also opened my windows defender firewall to Public. Any help appreciated. I am sure I may have missed something.

**Comment by mike12345567** [comment, marked as answer]
Hi @Townieeee - I responded to your Issue - find the response here: https://github.com/Budibase/budibase/issues/13563
